### ISSUE TITLE
Problem: eth_newBlockFilter is not working correct

### ISSUE THREAD

**Issue by StanislavKozlovsky** [crypto-org-chain/cronos#856]
**Looks like cronos using old version of ethermint.**

**In my case eth_newBlockFilter is not working correct.**
1. I am creating new filter in such way:
`curl -X POST --data '{"jsonrpc":"2.0","method":"eth_newBlockFilter","params":[],"id":1}' -H "Content-Type: application/json" my_url
{"jsonrpc":"2.0","id":1,"result":"0x2c0f61ad4ea4510c54029c4127c1560e"}`
2. Then I am getting eth_getFilterChanges
`curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getFilterChanges","params":["0x2c0f61ad4ea4510c54029c4127c1560e"],"id":1}' -H "Content-Type: application /json" my_url
{"jsonrpc":"2.0","id":1,"result":["0xdbd671f567cce86e1ca0a25660c0b5ae79d84a05fc1cae18e21d73c0271f792f",
.......]}`
3. However, if I take any hash from the eth_getFilterChanges response and try to find that block in the node, I get:
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getBlockByHash","params":["0xdbd671f567cce86e1ca0a25660c0b5ae79d84a05fc1cae18e21d73c0271f792f", true],"id":1}' -H "Content-Type : application/json" my_url
{"jsonrpc":"2.0","id":1,"result":null}
```

**Expected behavior**
Result is not null.

**After a little research, I found that:**
not so long ago a bug in ethermint was fixed
json-rpc(filters) fix block hash on newBlock filter #1503
https://github.com/evmos/ethermint/pull/1503/files
But I believe that this fix is not included in Cronos. 
Could you check please?

my node is fully synced. version: https://github.com/crypto-org-chain/cronos/releases/download/v1.0.2/cronos_1.0.2_Linux_x86_64.tar.gz

**Comment by yihuang** [comment]
yeah, we are using a fork to only cherry-pick non-consensus-breaking changes, it seems this one is not included, thanks for reporting, will cherry-pick it

**Comment by yihuang** [comment]
Closed by #857
